smoke-tests-otel-starter/src/test/java/io/opentelemetry/smoketest/OtelSpringStarterSmokeTest.java

```diff
@@ -56,28 +73,82 @@ class OtelSpringStarterSmokeTest {
 
   @Autowired private TestRestTemplate testRestTemplate;
 
-  @Autowired private ConfigProperties configProperties;
+  @Autowired private Environment environment;
+  @Autowired private PropagationProperties propagationProperties;
+  @Autowired private OtelResourceProperties otelResourceProperties;
+  @Autowired private OtlpExporterProperties otlpExporterProperties;
 
   @Configuration(proxyBeanMethods = false)
   static class TestConfiguration {
+
+    @Bean
+    ConfigurableMetricExporterProvider otlpMetricExporterProvider() {
+      return new ConfigurableMetricExporterProvider() {
+        @Override
+        public MetricExporter createExporter(ConfigProperties configProperties) {
+          return METRIC_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
+    }
+
     @Bean
-    public MetricExporter metricExporter() {
-      return METRIC_EXPORTER;
+    ConfigurableSpanExporterProvider otlpSpanExporterProvider() {
+      return new ConfigurableSpanExporterProvider() {
+        @Override
+        public SpanExporter createExporter(ConfigProperties configProperties) {
+          return SPAN_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
     }
 
     @Bean
-    public SpanExporter spanExporter() {
-      return SPAN_EXPORTER;
+    ConfigurableLogRecordExporterProvider otlpLogRecordExporterProvider() {
+      return new ConfigurableLogRecordExporterProvider() {
+        @Override
+        public LogRecordExporter createExporter(ConfigProperties configProperties) {
+          return LOG_RECORD_EXPORTER;
+        }
+
+        @Override
+        public String getName() {
+          return "memory";
+        }
+      };
     }
 
+    // this bean gets loaded here:
```

Comment:
Could you please move this comment to `OpenTelemetrySdkComponentLoader`

Reply:
what should the comment be then?

Because the SDK calls the loader in several places:

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/2832627/162e97b2-0b8e-4e51-9cfb-b60961b40630)

Reply:
Here https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8f18979bbe8a89abbac4a3b6c002d4a5a58f7a83/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java#L124